docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,403 @@
+
+After completing the above work, Base64 encode the `PublicLayerBlockCipherOptions` and annotation packet, and write them into the manifest.json of the container image as the annotation field of the layer, the former is placed in the `org.opencontainers.image.enc.pubopts` field and the latter in the `org.opencontainers.image.enc.keys.provider.attestation-agent` field. Finally, add the "+encrypted" suffix to the mediaType field of this layer. For example, the container image manifest with an encrypted layer is as follows:
+
+```json
+{
+    "schemaVersion": 2,
+    "config": {
+        "mediaType": "application/vnd.oci.image.config.v1+json",
+        "digest": "sha256:ef87d9f0...",
+        "size": 1510
+    },
+    "layers": [
+        {
+            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip+encrypted",
+            "digest": "sha256:f69ae40...",
+            "size": 204812,
+            "annotations": {
+                "org.opencontainers.image.enc.pubopts": "eyJjaXBoZXIiOi...",
+                "org.opencontainers.image.enc.keys.provider.attestation-agent": "eyJwcm9..."
+            }
+        }
+    ]
+}
+```
+
+### Image signing
+
+The container image signing standard used by the confidential containers V1 is follow the signing mechanism used in [containers/image](https://github.com/containers/image). At present, the only defined cryptographic signature format is a OpenPGP signature (RFC 4880). The signing process will not modify any content of the container image, including layer, config, manifest, etc. Therefore, the signature will not change the image manifest digest used to identify the image.
+
+The container signature under this standard is essentially a blob, which consists of two parts:
+
+- JSON payload
+
+- Cryptographic signature of JSON payload
+
+The JSON payload contains the manifest digest and identity information of the container image. Examples are as follows:
+
+```json
+{
+    "critical": {
+        "type": "atomic container signature",
+        "image": {
+            "docker-manifest-digest": "sha256:817a12c32a39be..."
+        },
+        "identity": {
+            "docker-reference": "docker.io/library/busybox:latest"
+        }
+    },
+    "optional": {
+        "creator": "some software package v1.0.1-35",
+        "timestamp": 1483228800,
+    }
+}
+```
+
+The process of signing a container image is as follows:
+
+1. Create a signature JSON payload and write the manifest digest together with some otheridentity information of the container image.
+2. Use the private key to sign the JSON payload generated in the previous step, and package the cryptographic signature together with the JSON payload into a signature claim file.
```

Comment:
Your signature claim files directory structure below only shows the signature file, but not the JSON payload. Is that intentional?

Reply:
The signature claim file contains JSON payload, cryptographic signature itself and some other signature information.